src/lib/functions/server.js

```diff
@@ -37,13 +37,23 @@ const buildClientContext = function (headers) {
   }
 }
 
+// same as https://github.com/netlify/bitballoon/blob/fbd7881e6c8e8c48e7a0145da4ee26090c794108/app/models/deploy.rb#L482
+// eslint-disable-next-line unicorn/better-regex
+const isValidNetlifyFunctionName = (name) => /^[A-Za-z0-9_-]+$/.test(name)
```

Comment:
We have a class that represents a function (see https://github.com/netlify/cli/blob/main/src/lib/functions/netlify-function.js). Rather than having the validation logic in the server, what do you think about moving it to the `NetlifyFunction` class and have a `isValidName` instance method?

Reply:
addressed in 80749db